chapter_recurrent-neural-networks/sequence.ipynb

```diff
@@ -0,0 +1,584 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%mavenRepo snapshots https://oss.sonatype.org/content/repositories/snapshots/\n",
+    "\n",
+    "%maven ai.djl:api:0.10.0-SNAPSHOT\n",
+    "%maven org.slf4j:slf4j-api:1.7.26\n",
+    "%maven org.slf4j:slf4j-simple:1.7.26\n",
+    "\n",
+    "%maven ai.djl.mxnet:mxnet-engine:0.10.0-SNAPSHOT\n",
+    "%maven ai.djl.mxnet:mxnet-native-auto:1.7.0-backport"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load ../utils/plot-utils\n",
+    "%load ../utils/Functions.java"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import java.util.Arrays;\n",
+    "\n",
+    "import tech.tablesaw.plotly.traces.*;\n",
+    "import tech.tablesaw.plotly.components.*;\n",
+    "\n",
+    "import ai.djl.ndarray.*;\n",
+    "import ai.djl.ndarray.types.*;\n",
+    "import ai.djl.ndarray.index.*;\n",
+    "import ai.djl.nn.*;\n",
+    "import ai.djl.nn.core.*;\n",
+    "import ai.djl.training.*;\n",
+    "import ai.djl.training.dataset.*;\n",
+    "import ai.djl.training.initializer.*;\n",
+    "import ai.djl.training.loss.*;\n",
+    "import ai.djl.training.optimizer.*;\n",
+    "import ai.djl.training.tracker.*;\n",
+    "import ai.djl.training.listener.*;\n",
+    "import ai.djl.translate.*;\n",
+    "import ai.djl.inference.*;\n",
+    "import ai.djl.Model;"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "NDManager manager = NDManager.newBaseManager();"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "public Figure plot(double[] x, double[] y, String xLabel, String yLabel) {\n",
+    "    ScatterTrace trace = ScatterTrace.builder(x,y)\n",
+    "        .mode(ScatterTrace.Mode.LINE)\n",
+    "        .build();\n",
+    "\n",
+    "    Layout layout = Layout.builder()\n",
+    "            .showLegend(true)\n",
+    "            .xAxis(Axis.builder().title(xLabel).build())\n",
+    "            .yAxis(Axis.builder().title(yLabel).build())\n",
+    "            .build();\n",
+    "\n",
+    "    return new Figure(layout, trace);\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "int T = 1000; //Generate a total of 1000 points\n",
+    "NDArray time = manager.arange(1f, T+1);\n",
+    "NDArray x = time.mul(0.01).sin().add(\n",
+    "    manager.randomNormal(0f, 0.2f, new Shape(T), DataType.FLOAT32));\n",
+    "\n",
+    "double[] xAxis = Functions.floatToDoubleArray(time.toFloatArray());\n",
```

Comment:
just out of curiosity, if it uses float, the graph is inaccurate?

Reply:
Tablesaw doesn't support float, only double. https://www.javadoc.io/static/tech.tablesaw/tablesaw-jsplot/0.38.1/tech/tablesaw/plotly/traces/ScatterTrace.html#builder-double:A-double:A-